src/select/typeahead/TypeaheadSelect.tsx

```diff
@@ -187,6 +189,19 @@ function TypeaheadSelect({
     }
   }
 
+  function handleKeyDown(event: React.KeyboardEvent<HTMLInputElement>) {
+    const {key} = event;
+
+    if (
+      key === KEYBOARD_EVENT_KEY.ESCAPE &&
+      !computedDropdownOptions.length &&
```

Comment:
I assume this line aims to prevent removing tags in case user had entered a part of keyword and trying to remove a character.
But I think this won't work if user entered a keyword, but there are no matching options (`!computedDropdownOptions.length` is `true`). In this case if user wants to remove the last character of the search keyword, he will end up removing the tag instead.

It may be better to check the input value state directly:
(Also I believe we should check if there are any visible tags, otherwise we don't need to run this block)
```suggestion
      !typeaheadInputValue &&
      tags.length
```